state/upgrades.go

```diff
@@ -1884,3 +1885,54 @@ func UpdateInheritedControllerConfig(pool *StatePool) error {
 	}
 	return nil
 }
+
+// EnsureDefaultModificationStatus ensures that there is a modification status
+// document for every machine in the statuses.
+func EnsureDefaultModificationStatus(pool *StatePool) error {
+	st := pool.SystemState()
+	db := st.db()
+
+	machineCol, machineCloser := db.GetRawCollection(machinesC)
+	defer machineCloser()
+	// Find all machines that are alive.
+	machineIter := machineCol.Find(bson.D{{"life", life.Alive}}).Iter()
+	defer machineIter.Close()
+
+	statusCol, statusCloser := db.GetRawCollection(statusesC)
+	defer statusCloser()
+
+	var ops []txn.Op
+	var machine machineDoc
+	for machineIter.Next(&machine) {
+		key := machineGlobalModificationKey(machine.Id)
```

Comment:
This is the wrong key as it doesn't include the model uuid. This is necessary because we are using raw collections.

Your QA procedures wouldn't have caught this because of the fix I landed yesterday in #9952. You'd need to look at the output of YAML status to ensure that the docs aren't missing.